maintainers/teams: add figsoda, tjni, and winter to rust

winterqt opened the pull request:
###### Description of changes

The three of us have been involved in most Rust-related PRs for a bit
now, so I think it makes sense to add us to the team.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

winterqt committed:
maintainers/teams: add figsoda, tjni, and winter to rust

The three of us have been involved in most Rust-related PRs for a bit
now, so I think it makes sense to add us to the team.

winterqt committed:
rustc, cargo: add rust team to maintainers

winterqt commented:
Not sure if I should add @figsoda and I to CODEOWNERS. @zowoq?

zowoq commented on pkgs/development/compilers/rust/cargo.nix:
```diff
@@ -76,7 +76,7 @@ rustPlatform.buildRustPackage {
     homepage = "https://crates.io";
     description = "Downloads your Rust project's dependencies and builds your project";
-    maintainers = with maintainers; [ retrry ];
+    maintainers = with maintainers; [ retrry ] ++ teams.rust;
```
```suggestion
    maintainers = with maintainers; [ retrry ] ++ teams.rust.members;
```

zowoq commented on pkgs/development/compilers/rust/rustc.nix:
```diff
@@ -211,7 +211,7 @@ in stdenv.mkDerivation rec {
     homepage = "https://www.rust-lang.org/";
     description = "A safe, concurrent, practical language";
-    maintainers = with maintainers; [ cstrahan globin havvy ];
+    maintainers = with maintainers; [ cstrahan globin havvy ] ++ teams.rust;
```
```suggestion
    maintainers = with maintainers; [ cstrahan globin havvy ] ++ teams.rust.members;
```

zowoq commented:
> Not sure if I should add figsoda and I to CODEOWNERS. 

Feel free. 

https://github.com/NixOS/nixpkgs/commit/8ce6fdf581b191d67c62590ec73a22cbf8797ae8 I removed myself as the only codeowner for the rust packaging and docs but if it was going to be a couple of us we could add that back as well?

zowoq commented on maintainers/team-list.nix:
```diff
@@ -699,8 +699,11 @@ with lib.maintainers; {
   rust = {
     members = [
       andir
```
Don't think I knew that this team existed, must have been generated from codeowners.

Please remove `andir`, they removed themselves treewide in 31e5b8dc2198eacec3e009e21fc8226486aa674e, I think they're only listed here because they used to be a codeowner.

zowoq commented:
Made a rust github team: https://github.com/orgs/NixOS/teams/rust

winterqt committed:
CODEOWNERS: add new Rust team members, re-add Rust build support/docs

8ce6fdf581b191d67c62590ec73a22cbf8797ae8 removed the build support/docs
entries, but per discussion in #213697, we think having multiple members
justifies its readdition.

winterqt committed:
maintainers/teams: remove andir from rust

andir removed himself as a maintainer treewide in
31e5b8dc2198eacec3e009e21fc8226486aa674e, but seemingly missed this entry.

winterqt commented:
> [8ce6fdf](https://github.com/NixOS/nixpkgs/commit/8ce6fdf581b191d67c62590ec73a22cbf8797ae8) I removed myself as the only codeowner for the rust packaging and docs but if it was going to be a couple of us we could add that back as well?

Wish you added some sort of justification to that commit message 😕

zowoq commented:
> Wish you added some sort of justification to that commit message 😕

It doesn't matter.
